src/main/kotlin/com/exactpro/th2/kafka/client/KafkaConnection.kt

```diff
@@ -58,10 +59,11 @@ class KafkaConnection(
     fun publish(message: RawMessage) {
         val alias = message.sessionAlias ?: error("Message '${message.id.logId}' does not contain session alias.")
         val value = message.body.toByteArray()
+        val bookConfig = config.booksConfigs[message.bookName] ?: config.defaultBookConfig
         val messageIdBuilder = message.id.toBuilder().apply {
             direction = Direction.SECOND
-            bookName = factory.boxConfiguration.bookName
-            setConnectionId(connectionIdBuilder.setSessionGroup(config.aliasToSessionGroup.getValue(alias)))
+            bookName = message.bookName
+            setConnectionId(connectionIdBuilder.setSessionGroup(bookConfig.aliasToSessionGroup.getValue(alias)))
```

Comment:
Please execute check before building message and report understandable error.
Also, look at the attached schema. It describe what book should be used for error event depends on parent event id 
https://jamboard.google.com/d/1cB_X0iuvcHGvz9_mgaR7u_VL6T1Ee_tkI-Ljmi4rUXU/viewer?f=0